### DISCUSSION TITLE
A question about the two different loss values in VITS

### DISCUSSION THREAD

**Discussion by rioharper** [coqui-ai/TTS#1053] [General Q&A] [answered]
I've been playing around with your technology for a while, but I was confused when finetuning a VITS model for the first time, as it seems to have two separate loss values, both of which communicate very different narratives about training. I have a dataset of about an hour 15 minutes. Tensorboard: 
![image](https://user-images.githubusercontent.com/50555644/147995021-23b1f69d-e860-4470-a053-95bf53d74a94.png)

So are they evaluating different parts of my model? Using different metrics? Is there any one of them I should look more closely too? My audio samples substantially improve much past loss 0 shows overfitting. Ive tried this out with different datasets, but I have seen the same issue. Thanks!

Config:
audio_config = BaseAudioConfig(
    sample_rate=22050,
    win_length=1024,
    hop_length=256,
    num_mels=80,
    preemphasis=0.99,
    ref_level_db=2,
    log_func="np.log",
    do_trim_silence=True,
    trim_db=45,
    mel_fmin=0,
    mel_fmax=None,
    spec_gain=1.0,
    signal_norm=False,
    do_amp_to_db_linear=False,
)

config = VitsConfig(
    audio=audio_config,
    run_name="vits_ljspeech",
    batch_size=32,
    eval_batch_size=16,
    batch_group_size=5,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="english_cleaners",
    use_phonemes=True,
    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    compute_input_seq_cache=True,
    print_step=25,
    print_eval=True,
    mixed_precision=False,
    max_seq_len=500000,
    output_path=output_path,
    datasets=[dataset_config],
)

**Comment by rioharper** [comment, marked as answer]
nevermind, I found that vits_TrainEpochStats/avg_loss_1 is the best metric to see loss.

**Reply by davidmartinrius** [reply]
I agree with you. After multiple experiments, when the model starts overfitting the TrainEpochStats/avg_loss_1 is the only indicator that shows a divergence and the value increases. I don't know if it is the best indicator but it worked for me. I guess that at this point I should start a fine tuning of the model with more fresh data..
![Screenshot from 2022-06-07 23-47-39](https://user-images.githubusercontent.com/16558194/172488913-5a4014c6-7d80-4bf5-92bd-4d6adebe5a13.png)
.
